### PR TITLE
#181-exception-when-otp-is-null

### PR DESCRIPTION
**Pull request by IgorMaksymov**
Exception is generated, when the  one time password is incorrect
connect to #181

### CODE REVIEW

**Comment by IgorMaksymov** [comment]
conncect to #181

**Comment by vaa25** [comment]
@IgorMaksymov @AndriyBaibak Can you describe more extensively this bug? I cant understand what about talks here.

File: restservice/src/test/java/org/itevents/service/transactional/MyBatisUserServiceTest.java [hunk 1]
```diff
@@ -294,4 +294,12 @@ public void shouldNotUseOtpIfExpired() throws Exception {
 
         userService.activateUserWithOtp(stringOtp);
     }
+
+    @Test(expected = EntityNotFoundServiceException.class)
+    public void shouldThrowServiceExceptionIfOtpIsInvalid() throws Exception {
```
Comment (vaa25) [inline restservice/src/test/java/org/itevents/service/transactional/MyBatisUserServiceTest.java]: @IgorMaksymov What tests this test?

**Comment by romach** [comment]
* PASSED @AndriyBaibak 30 minutes

**Comment by vaa25** [comment]
@IgorMaksymov you forgot to log exception

**Comment by IgorMaksymov** [comment]
@vaa25 is this so important, to log a mistype?
@alex-anakin we need to log all of our exceptions?

**Comment by vaa25** [comment]
@IgorMaksymov 
almost PASSED. You can merge your pr after merging master in your branch and checking all ok.

**Comment by IgorMaksymov** [comment]
@vaa25 there is no conflicts with master, it will merge it without any conflicts.

**Comment by IgorMaksymov** [comment]
@vaa25 do not merge it!

**Comment by IgorMaksymov** [comment]
@vaa25 its OK now.

File: restservice/src/main/java/org/itevents/service/transactional/MyBatisUserService.java [hunk 2]
```diff
@@ -168,16 +168,19 @@ public void setOtpToUser(User user, OneTimePassword oneTimePassword) {
...
+
+            if (oneTimePassword.getExpirationDate().after(new Date()) ) {
+                user.setRole(roleService.getRoleByName("subscriber"));
+                userDao.updateUser(user);
+            } else {
+                String message = "Password expired";
+                throw new OtpExpiredServiceException(message);
+            }
+        } catch (EntityNotFoundDaoException e) {
+            throw new EntityNotFoundServiceException(e.getMessage(),e);
```
Comment (IgorMaksymov) [inline restservice/src/main/java/org/itevents/service/transactional/MyBatisUserService.java]: @vaa25 look here pls

**Comment by vaa25** [comment]
@IgorMaksymov create todo at `new Date()` and refer to #195 . then can merge
@AndriyBaibak 10 min

**Pull request merged**